Updating menu files in Russian

RestorerZ opened the pull request:
Updating menu files in Russian
Обновление файлов меню на русском языке
File description in Russian of the directory contents
Файл описания на русском языке содержимого каталога

RestorerZ committed:
Updating menu files in Russian

Updating menu files in Russian
Обновление файлов меню на русском языке

RestorerZ committed:
Merge pull request #1 from RestorerZ/RestorerZ-patch-1

Updating menu files in Russian

RestorerZ committed:
File description in Russian

File description in Russian of the directory contents
Файл описания на русском языке содержимого каталога

RestorerZ committed:
Merge pull request #2 from RestorerZ/RestorerZ-patch-2

File description in Russian

RestorerZ committed:
Updated and corrected translation messages

Updated and corrected translation of program messages

Дополненный и исправленный перевод сообщений программы

RestorerZ committed:
Merge pull request #3 from RestorerZ/RestorerZ-patch-3

Updated and corrected translation messages

brammool commented:
Please run the check script, as documented in src/po/README.  There are a few errors:
"ru.po" 11399 lines, 424179 bytes
1890Mismatching % in line 6317
1891msgid: 
1892msgstr: %%
1893Mismatching % in line 8247
1894msgid: %%%d%%%%
1895msgstr: %d%%%%%%
1896Mismatching % in line 9424

RestorerZ commented:
> Please run the check script, as documented in src/po/README. There are a few errors:
> "ru.po" 11399 lines, 424179 bytes
> 1890Mismatching % in line 6317
> 1891msgid:
> 1892msgstr: %%
The original message reads as:
msgid ""
"VIMRUN.EXE not found in your $PATH.\n"
"External commands will not pause after completion.\n"
"See  :help win32-vimrun  for more information."
However, in Windows, environment variables are usually denoted in percent signs.
And, accordingly, the PATH environment variable is specified as %PATH% in the translation
msgstr ""
"Не удалось найти файл VIMRUN.EXE по тем маршрутам,\n"
"которые указаны в переменной окружения %PATH%.\n"
"По этой причине после исполнения внешних команд,\n"
"вызванных из программы GVim, окно командной оболочки\n"
"будет закрыто автоматически.\n"
"Чтобы получить дополнительную информацию, наберите   :help win32-vimrun"

В оригинальном сообщении указано как:
msgid ""
"VIMRUN.EXE not found in your $PATH.\n"
"External commands will not pause after completion.\n"
"See  :help win32-vimrun  for more information."
Однако в ОС Windows переменные окружения принято обозначать в знаках процента.
И, соответственно, переменная окружения PATH указана как %PATH% в переводе
msgstr ""
"Не удалось найти файл VIMRUN.EXE по тем маршрутам,\n"
"которые указаны в переменной окружения %PATH%.\n"
"По этой причине после исполнения внешних команд,\n"
"вызванных из программы GVim, окно командной оболочки\n"
"будет закрыто автоматически.\n"
"Чтобы получить дополнительную информацию, наберите   :help win32-vimrun"

> 1893Mismatching % in line 8247
> 1894msgid: %%%d%%%%
> 1895msgstr: %d%%%%%%
The original message reads as:
msgid "%a %b %d %H:%M:%S %Y"
But the Russian date and time format prescribes a different sequence, as specified in the translation:
msgstr "%d %b %Y %a %H:%M:%S"
moreover, this translation of the message exists since the Vim 8.1 version and has not been changed by me.

В оригинальном сообщении указано как:
msgid "%a %b %d %H:%M:%S %Y"
Но русский формат даты и времени предписывает иную последовательность, так как указано в переводе:
msgstr "%d %b %Y %a %H:%M:%S"
причём данный перевод сообщения существует ещё с версии Vim 8.1 и мной не менялся.

> 1896Mismatching % in line 9424
The original message reads as:
msgid "E1035: % requires number arguments"
In translation, the sign of the operation of taking the remainder of the division ( % ) is described in words so that it is more clear and does not get out of the common context of the translation
msgstr ""
"E1035. Тип переменных должен быть Number для выполнения операции деление с "
"остатком"

В оригинальном сообщении указано как:
msgid "E1035: % requires number arguments"
В переводе знак операции взятие остатка от деления (%) описан словами, чтобы было более понятно и не выбивалось из общего контекста перевода
msgstr ""
"E1035. Тип переменных должен быть Number для выполнения операции деление с "
"остатком"

chrisbra commented:
> %PATH%

I would leave as $PATH, since the translation is valid for Unix as well as windows.

The one ones might require the 

# no-c-format 

flag/comment (not sure, I do not completely understand the problem)

RestorerZ commented:
> I would leave as $PATH, since the translation is valid for Unix as well as windows.
This message is only used in the Windows version. This is shown not only by the file names in this message, but also by [this](https://github.com/vim/vim/blob/81e17fbe00fd2ed93b262adc3ba41c86b02e7f46/src/os_win32.c#L4967)

Это сообщение используется только в версии для ОС Windows. На это указывает не только наименования файлов в этом сообщении, но и [это](https://github.com/vim/vim/blob/81e17fbe00fd2ed93b262adc3ba41c86b02e7f46/src/os_win32.c#L4967)

> The one ones might require the
> # no-c-format
Yes, when you add this flag, there will be no error.

Да, при добавлении этого флага, ошибки не будет.

I hope you agree with me that the program messages should be usual and correspond to the system where the user receives them.

Надеюсь, вы со мной согласитесь, что сообщения программы должны быть привычны и соответствовать той системе, где пользователь их получает.

chrisbra commented:
> I hope you agree with me that the program messages should be usual and correspond to the system where the user receives them.

I do not. Since the way vim handles environment variables is by using the `$PATH` as a unified way to notate environment variables (independently of the system where it runs on).

RestorerZ commented:
> I do not. Since the way vim handles environment variables is by using the `$PATH` as a unified way to notate environment variables (independently of the system where it runs on).

Yes, I completely agree with you. The Vim program handles environment variables as $PATH, but this is its internal representation (as we say "kitchen").
I'm talking about what is shown to the user in the message. When a Windows user sees this message and Googles $PATH, they won't find what they were looking for. More precisely, it will find, but not immediately. You and I know what $PATH is, but what if it is some inexperienced user or a novice in Viming?
I hope you understand what I want to say, despite my poor English.

Да, я с вами полностью согласен. Программа Vim обрабатывает переменные окружения как $PATH, но это её внутреннее представление (как говорят у нас "кухня").
Я же говорю о том, что показывается пользователю в сообщении. Когда пользователь ОС Windows увидит это сообщение и будет гуглить $PATH, то он не найдёт, то что искал. Точнее найдёт, но не сразу. Мы с вами знаем, что такое $PATH, а если это будет какой-нибудь неопытный пользователь или новичок в виминге?
Надеюсь, что вы понимаете, что я хочу сказать, несмотря на мой плохой английский язык.

tonymec commented:
Let's be pragmatic.

There are many places where Vim names environment variables in cros-platform messages, and there they are always displayed as $NAME, which is, hall we say, "the Vim way" of naming environment variables. One of these variables is $VIMRUNTIME, another is $VIM; they are used on all platforms, and Vim is not going to start naming them %VIMRUNTIME% and %VIM% on Vim only to please, historically, the developers of MS-DOS 2.0 who decided to name their environment variables otherwise than on Unix.

For consistency, IMHO it wouldn't be "abnormal" to use this same "Vim terminology" even in messages which will only be displayed on Windows — I think it will help newbie users to get a "feeling" for the "different" way Vim names things (in contrast to Windows's usual, and OTOH in line with usual Unix terminology).

Best regards,
Tony.

tonymec commented:
oops, I meant "Vim is not going to start naming them %VIMRUNTIME% and %VIM% on **Windows** only to please" etc.

Best regards,
Tony.

RestorerZ commented:
Yes, it is for pragmatic reasons that I proceed.
Let's leave the debate as to which method of naming environment variables is preferable - - - UINIX-like, Windows-like, or CP/M-like, or some other system.
When you type `:e $MYVIMRC` in Vim using the $ symbol for the environment variable, you do it in the editor itself, and you can do it any way you want, because these are the internal rules of the editor, regardless of what system it is running on.
But when you talk about the environment variables of the system itself, it is correct, in my firm belief, to say it in terms of the system in which you are located.
For example, you won't tell an OpenVMS user to type the path to their file as /home/user/docs/Letter.txt ? In his system, it looks alien and not right.
In common, I don't think our discussion is worth spending so much time on. There are certain rules and traditions adopted in each system, and they should be respected and tried to follow. Do you agree with me?

Да, именно из прагматических соображений я исхожу.
Оставим споры, какой способ именования переменных окружения предпочтителен --- UINIX-подобный, Windows-подобный или CP/M-подобный, или какой-либо другой системы.
Когда вы в Vim набираете `:e $MYVIMRC` используя для этого обозначение переменной окружения символ $, вы делаете это в самом редакторе и там можно это делать как угодно, потому что это внутренние правила редактора независящее от того в какой системе он работает.
Но когда вы говорите о переменных окружения самой системы, то правильно, по моему твёрдому убеждению, говорить это в терминах той системы, в которой вы находитесь.
Вы же, например, не будете говорить пользователю OpenVMS, что он должен набирать путь к своему файлу как /home/user/docs/Letter.txt ? В его системе это выглядит чужеродно и не правильно.
А вообще, мне кажется, наша дискуссия того не стоит, чтобы тратить на неё столько времени. Есть определённые правила и традиции, принятые в каждой системе, и их стоит уважать и стараться соблюдать. Вы со мной согласны?

brammool commented:
Translations should never change the semantics of the message, only the language.  If some construct differs between systems this requires another message, not another translation.  So $PATH should never be change to anything else.

RestorerZ commented:
The translation is intended to convey the essence of the message most accurately in a language that the user understands, without changing the meaning of the message. Yes, I fully agree with you on this.
And that is why the translation did not change the meaning, essence, or concept of the PATH environment variable in any way. And this essence is as follows:
[On POSIX and Unix-like operating systems, the $PATH variable is specified as a list of one or more directory names separated by colon (:) characters.](https://en.wikipedia.org/wiki/PATH_(variable))
[On DOS, OS/2, and Windows operating systems, the %PATH% variable is specified as a list of one or more directory names separated by semicolon (;) characters.](https://en.wikipedia.org/wiki/PATH_(variable))
But, the thing that in the original message and translation, the designation of this environment variable differs, Yes, it is bad, and should be removed.
I will change the notation in the translation to what is in the original message.
Will you check the notation in the English message?

Перевод призван наиболее точно передавать суть сообщения на языке понятном пользователю, не изменяя смысл сообщения, да, в этом я полностью согласен с вами.
И именно поэтому в переводе никоим образом не менялся смысл, суть, понятие переменной окружения PATH. И эта суть в следующем:
[На POSIX и Unix-подобных операционных системах переменная $PATH задана как одно или несколько имён каталогов, разделённых двоеточиями (:).](https://ru.wikipedia.org/wiki/PATH_(%D0%BF%D0%B5%D1%80%D0%B5%D0%BC%D0%B5%D0%BD%D0%BD%D0%B0%D1%8F))
[В операционных системах DOS, OS/2 и Windows переменная %PATH% задана в виде списка каталогов, разделённых точкой с запятой (;).](https://ru.wikipedia.org/wiki/PATH_(%D0%BF%D0%B5%D1%80%D0%B5%D0%BC%D0%B5%D0%BD%D0%BD%D0%B0%D1%8F))
А вот, то, что в оригинальном сообщении и переводе обозначение этой переменной окружения различаются, да, плохо, и должно быть утсранено.
Я изменю пока в переводе обозначение на то, что в оригинальном сообщение.
Вы проверите обозначение в английском сообщении?

RestorerZ committed:
Changing the notation of the PATH

Change the notation of the PATH environment variable until the original message is corrected
Изменение обозначения переменной среды PATH до исправления исходного сообщения

RestorerZ committed:
Russian language strings for gvim NSIS installer

russian.nsi: Russian language strings for gvim NSIS installer.

RestorerZ committed:
Added the file name russian.nsi

Added the file name russian.nsi
russian.nsi: Russian language strings for gvim NSIS installer

brammool commented:
The pull request deletes gvim.nsi. that looks like a mistake.
menu_ru_ru.vim was maintained by Sergey Alyoshin, did you communicate with him?

RestorerZ commented:
For some reason, the file was replaced instead of merging. Although the comparison clearly shows the difference in one line. I'll try again.
Yes, Sergey and I communicated. I explained.

По какой-то причине файл был заменен вместо слияния. Хотя сравнение ясно показывает разницу в одну линию. Я попробую еще раз.
Да, мы с Сергеем разговаривали. Я объяснил ему.

ArturAmpilogov commented on src/po/ru.po:
```diff
@@ -3,2415 +3,3412 @@
 msgid "E831: bf_key_init() called with empty password"
-msgstr "E831: bf_key_init() вызван с пустым паролем"
+msgstr ""
+"E831. Внутренняя функция bf_key_init(). Вызов с нулевым значением аргумента"
```
Hi @RestorerZ , 
It looks there are some issues with the translation.
1) The message `:` symbol is replaced with `.` everywhere in msgstr.
2) The previous translation is more precise and compact.
+Sergey Alyoshin for review.

RestorerZ replied:
@ArturAmpilogov, thank you for your feedback. I replaced the full stop after the error
number with a colon. Although this is not quite correct according to the rules
of the Russian language, and compatibility is more important.
As for the example you provided.  You know what the thing is, the function can't
be called with an empty password. However, you can't call it with a full
password either.  The function can be called with some argument value. But an
empty value can be passed to the argument of this function. In this case, the
password value.
You know that without me, right?
And, as a Russian speaker, you understand that bf_key_init() is not called, but called.
The word "function" is a feminine noun in Russian.
But this is so, small things. That's just the little things there...

P. S.
A, and yes, if you really put the pros for the translation, then these pros
should be put to Vasily Ragozin, who did almost the entire translation.
But Alexey Alyoshin should be put on the plus side for the fact that the plural
appeared in the translation and now you don't have to be perverted, coming up
with how best to translate words with counters.
Anyway, I heard you. You would like the phrases to be more succinct and shorter.
I'll try to take this into account.

Артур, благодарю за ваш отзыв. Я заменил точку после номера ошибки на
двоеточие, хоть это и не совсем правильно по правилам русского языка, но
совместимость прежде всего.
Что касается приведённого вами примера, то понимаете какая штука, функцию нельзя
вызвать с пустым паролем. Впрочем и с полным паролем её тоже вызвать нельзя.
Функцию можно вызвать с каким‐то значением аргумента. А вот в аргумент этой
функции и может быть передано пустое значение. В данном случае значение пароля.
Вы, впрочем, это знаете и без меня, так ведь.
И, как русскоговорящий, вы ведь понимаете, что bf_key_init() не вызван, а вызванА.
Слово "функция" - существительное женского рода в русском языке.
Но это так, мелочи. Вот только мелочей там... 

P. S.
А, и да, если вы уж ставите плюсы за перевод, то эти плюсы надо ставить Василию
Рагозину, который и сделал практически весь перевод.
А вот Алексею Алёшину плюсы надо ставить за то, что в переводе появилось
множественное число и теперь не приходится извращаться, придумывая как лучше
перевести слова со счётчиками.
В любом случае я услышал вас. Вы бы хотели, чтобы фразы были более ёмкие и короче.
Я постараюсь это учесть.

alyoshin replied:
> И, как русскоговорящий, вы ведь понимаете, что bf_key_init() не вызван, а вызванА.
> Слово "функция" - существительное женского рода в русском языке.

Вы нашли опечатку, спасибо. 


> Но это так, мелочи. Вот только мелочей там...

Пожалуйста, также конкретно как с предыдущей опечаткой.


> Василию Рагозину, который и сделал практически весь перевод.

Не умаляя заслуг Василия, я (кстати, Сергей) сопровождаю перевод интерфейса уже 7 лет. За это время перевод успел измениться.

alyoshin replied:
> Вы бы хотели, чтобы фразы были более ёмкие и короче.

Мы (не я один) бы хотели, чтобы перевод соответствовал оригинальному переводимому тексту.

chrisbra replied:
> @ArturAmpilogov, thank you for your feedback. I replaced the full stop after the error
number with a colon. 

That is not quite correct and might break a lot of scripts that test for the existence of an error in the catch case. So please do not change the punctuation. (What I means is replacing the `:` by `.`).

RestorerZ replied:
@alyoshin, рад, что вы откликнулись на сообщение.
Вы спрашивали, почему "аргумент" не является калькой с английского языка?
Думаю, что на этот вопрос вам лучше ответит русская математическая школа или
школьный учебник по алгебре в той части, где вводится понятие функции.  Это что
касается функций, как способа реализации решения определённого алгоритма.  Что
касается командной строки, то нужно понимать, что программа - это совокупность
функций и процедур, решающих какую-то задачу.  И, соответственно, сама программа
является как бы "надфункцией" (большой функцией), при вызове которой ей задаются
начальные значения через механизм аргументов.  А вот в процессе работы программы
вы можете влиять на её поведение через указание определённых параметров.
Вы знаете, у меня почему-то такое ощущение, что представленная мной редакция
перевода воспринимается вами как-то не так.  Расценивайте её как, ну...
редакторскую правку что ли. Тем более вы же видели преамбулу к данной редакции
перевода.

 Основные задачи, решаемые данной редакцией перевода:
   - Соответствие нормам и правилам русского языка;
   - Унифицированность сообщений;
   - Связанность сообщений;
   - Полнота и понятность сообщений без обращения к встроенной документации;
   - Лаконичность.

Если я ошибся в своих ощущениях, то рад этому. И тот вариант, который мной
предложен, тоже не лишён недостатков, неточностей, опечаток и ошибок. И это,
несмотря на то, что я проштудировал документацию и излазил код вдоль и поперёк в
части отображения того или иного сообщения, чтобы это самое сообщение не
противоречило действиям программы, и было понятно конечному пользователю. И я не
решусь сказать, что мой вариант это истина в последней инстанции и с готовностью
совместными усилиями доведу перевод до определённой степени совершенства,
независимо от того, чья это будет редакция. Но вот только я вам показываю со
ссылками, почему это так, а в ответ что-то про то, что так нельзя без каких-то
подкреплений.  Возможно, что вы не совсем знакомы с процессом локализации
программного обеспечения.
Но вы хотели примеров? Хорошо.
```
msgid "Latin supplement"
msgstr "Латынь дополненная"
msgid "Latin extended"
msgstr "Латынь расширенная"
```
Скажите, что это? Латынь - это мёртвый язык, который сейчас, в основном, широко
используется в медицине. Но я так понимаю, что редактор VIM стал поддерживать
этот язык и при вводе команды `:digraph!` он отображает какую-то "дополненную" и
"расширенную" латынь, язык древних римлян. Правильно? Нет. В данном случае речь
идёт о плоскостях кодового пространства Юникод. Для чего вы придумываете
какие-то новые понятия? Тем более ещё и неправильные. Ведь достаточно было
посмотреть в документации что это, зайти, например, на "Википедию" и переписать
от туда правильные, унифицированные наименования.
```
msgid "Greek and Coptic"
msgstr "Греческий и Коптский"
msgid "CJK symbols and punctuation"
msgstr "Китайские, Японские и Корейские символы и пунктуация"
```
А это? Вы разве не знаете, что в русском языке, в отличие от того же
английского, наименования языков пишутся со строчной буквы? Вы ведь
представляете этот перевод общественности. И об уровне переводчика как раз и
судят по таким вещам, в том числе и те, кто не является носителем языка, на
который выполнен перевод. Переводчик должен очень, очень хорошо знать родной
язык, и если в чём-то не уверен, то сверяться со словарями и справочниками.
```
msgid ""
"E864: \\%#= can only be followed by 0, 1, or 2. The automatic engine will be "
"used "
msgstr ""
"E864: после \\%#= может быть только 0, 1 или 2. Будет использоваться "
"автоматическая машина"
```
Скажите, что такое "автоматическая машина"? Это в смысле с АКПП? Ведь нет же. В
документации чёрным по белому написано, что в программе реализовано два
механизма, "движка" обработки регулярных выражений. И что при не правильном
указании значения данного параметра или его нулевом значении механизм обработки
будет выбираться автоматически, на основании определённых алгоритмов.
Или буквально следующий перевод:
```
msgid "Switching to backtracking RE engine for pattern: "
msgstr "Включено отслеживание движка рег. выражений для шаблона: "
```
Здесь вы уже используете понятие "движок", т. е. один и тот же термин
переводится по-разному, что нарушает принцип связанности сообщений, когда один и
тот же термин в одном и том же контексте должен переводиться одинаково. Но то
ладно. В этом переводе вообще явная ошибка. Во-первых, не "включено", а
выполнено "переключение" механизма обработки регулярных выражений для этого
конкретного поискового шаблона. Во-вторых, "backtrecking" - это наименование
механизма обработки регулярных выражений и на русский язык переводится как
"поиск с возвратом".
```
msgid "E973: Blob literal should have an even number of hex characters"
msgstr ""
"E973: блоб-литерал должен иметь чётное количество шестнадцатеричных символов"
```
А что такое "блоб-литерал"? Вы опять вводите какие-то свои понятия, которые
отсутствуют в программе и её документации? Вы хоть понимаете, что пользователь,
увидев сообщение в таком переводе, будет введён в заблуждение? В данном случае
речь идёт о типе данных BLOB являющихся единым, неделимым элементом
("единицей", "литералом") данного типа данных.
Сергей, продолжать с примерами можно очень долго, но и так уже много текста, да
и смысл, надеюсь, понятен.  Вы ведь знаете, что локализация - это не
"подстрочный перевод" исходного текста, а донесение до конечного пользователя
понятного ему смысла сообщения, наиболее близко соответствующего исходному
сообщению. Пообщайтесь с коллегами на специализированных форумах, почитайте
документацию и руководства по переводу интерфейсов программного сообщения.
Я могу предоставить вам все имеющиеся у меня материалы, и вы приведёте перевод в
соответствие с общепринятыми правилами и нормами. Хорошо?

RestorerZ replied:
Yes, @chrisbra, this is my mistake, I admit. My perfectionism played a cruel
trick on me. I'll keep that in mind for the future.

Да, chrisbra, это моя оплошность, признаю. Мой перфекционизм сыграл со мной злую
шутку. Я учту это на будущее.

alyoshin replied:
> Вы знаете, у меня почему-то такое ощущение, что представленная мной редакция
перевода воспринимается вами как-то не так.

Воспринимается как "один я умею переводить правильно". Особенно в связи с вашими невежливыми спотыканиями в учебники и пространными объяснениями. 


> Возможно, что вы не совсем знакомы с процессом локализации программного обеспечения.

Вполне знаком.


> Вы разве не знаете, что в русском языке, в отличие от того же английского, наименования языков пишутся со строчной буквы?

Здесь я согласен.


> Вы ведь представляете этот перевод общественности.

Ой, простите, как я посмел без вашего позволения.


> Скажите, что такое "автоматическая машина"? Это в смысле с АКПП?

Автомобильные аналогии? Если бы не ваша язвительность, вам бы не пришлось столько писать и в конце акцентировать на этом внимание.


> И об уровне переводчика как раз и судят по таким вещам, в том числе и те, кто не является носителем языка, на
который выполнен перевод. Переводчик должен очень, очень хорошо знать родной язык, и если в чём-то не уверен, то сверяться со словарями и справочниками.

Вы не очень вежливо общаетесь с незнакомым человеком.


> Здесь вы уже используете понятие "движок", т. е. один и тот же термин переводится по-разному

Я за использование одинаковых терминов, согласен.


> В этом переводе вообще явная ошибка. Во-первых, не "включено", а выполнено "переключение" механизма обработки регулярных выражений для этого конкретного поискового шаблона. 

В данном случае, мне кажется правильным "включен" а не "переключен". Как вы переведёте "switch light on"?


> Во-вторых, "backtrecking" - это наименование механизма обработки регулярных выражений и на русский язык переводится как "поиск с возвратом".

Вы опечатались, backtracking. Термины обсуждаемы. Но pattern в регулярном выражении по мне неплох и как "шаблон".


> А что такое "блоб-литерал"? Вы опять вводите какие-то свои понятия, которые отсутствуют в программе и её документации?

Я могу понять, что не стоит переводить BLOB, но вы предлагаете не переводить number, float, ...


> Пообщайтесь с коллегами на специализированных форумах, почитайте документацию и руководства по переводу интерфейсов программного сообщения. Я могу предоставить вам все имеющиеся у меня материалы, и вы приведёте перевод в соответствие с общепринятыми правилами и нормами. Хорошо?

Нет.

RestorerZ replied:
@alyoshin
Значит, мои ощущения меня не обманули, жаль. Жаль, что вы это так воспринимаете.
Я не думаю, что вы нуждаетесь в моём одобрении или неодобрении того, что вы
делаете. А тем более в разрешении на то, чтобы что-то сделать.  Так для чего
тогда вы это пишите? Пытаетесь меня подначить? Для чего?  Играете на публику?
Опять же вопрос, для чего?
Как правило, если аргументированно указывают на ошибку, то ошибку или исправляют,
или аргументируют, что, так как сделано это правильно. Всё остальное пустое.
А как высказанное мной моё личное мнение о навыках необходимых переводчику в
общем смысле связано с вежливым или не вежливым общением?

Ну, вообще-то
```
switching  n 
1. переключение, коммутация
Переключение, установление связи.
2. магн. перемагничивание
switching  v 
коммутация
Процесс маршрутизации путём переключения промежуточных соединений для установления связи между двумя узлами компьютерной сети.
switching  adj 
переключательный, переключающий
```
Сергей, но дело даже не в этом. Вы что, действительно не понимаете в чём там ошибка?
Это сообщение, скорее всего, выводится или при отладке, или при значении
'verbose' больше нуля. Т. е. это какое-то диагностическое сообщение.
Программа обрабатывала регулярные выражения, используя один из имеющихся у неё
механизмов, а для обработки какого-то конкретного шаблона выполнила переключение
на другой. Но уж точно никак не собирается следить, отслеживать "движок",
механизм обработки регулярных выражений для этого самого шаблона.

Я нигде не писал, что не надо переводить слова number, float, но если вы про
программу, то там речь идёт не просто о таких понятиях как "число" или "список", а о
наименованиях типов данных. Ведь не зря же они и в сообщениях и в документации
написаны с прописной буквы. А с наименованиями типов данных введённых автором
для их обозначения лучше не связываться и оставить так как это назвал автор. В
сопроводительной документации к программе можно, да и нужно, дать пояснение
пользователю, что есть русские аналоги этого наименования. И то, что этот тип
данных является, по сути, упорядоченным массивом или ассоциативным массивом. Но
вот переводить лучше не надо. Это поможет пользователю и при работе с
документацией и при общении с коллегами и при поиске в Интернет.

Ну, нет, так нет. А "нет" потому что вы займётесь редактурой своего перевода и у
вас уже есть необходимые материалы? Или "нет" потому что вы не считаете эти
нормы и правила общепринятыми и применимыми к вам и вашему переводу?

k-takata commented:
It seems that you changed the fileformat of `nsis/gvim.nsi` from `unix` to `dos`. It should be reverted.

The title of this PR is "Updating menu files in Russian", but it also includes `src/po/ru.po` and `nsis/gvim.nsi`. It is confusing. The title should be properly updated.
However, the problem is that you have already created a PR to update `src/po/ru.po` at #6734, but the current maintainer @alyoshin doesn't agree the changes.
So, I suggest to remove `src/po/ru.po` from this PR.

RestorerZ commented:
> It seems that you changed the fileformat of `nsis/gvim.nsi` from `unix` to `dos`. It should be reverted.

Yeah, that's why there was a mistake! Oh, those line endings. Yes, of course,
I'll fix it.
 
> The title of this PR is "Updating menu files in Russian", but it also includes `src/po/ru.po` and `nsis/gvim.nsi`. It is confusing. The title should be properly updated.
> However, the problem is that you have already created a PR to update `src/po/ru.po` at #6734, but the current maintainer @alyoshin doesn't agree the changes.
> So, I suggest to remove `src/po/ru.po` from this PR.

Well. Remove.

RestorerZ commented:
@k-takata 
The fact is that when I sent these pull requests, I still had a poor
understanding of how Git works (and even now I'm not very confident).
So here. I did something stupid. Instead of performing Fetch and Merge, I
deleted the repository and made a new Fork.  Accordingly, I can't update or
delete these files right now.
Tell me what I need to do. Write to Bram to delete these files? Or maybe you can
delete them?  And in pull request #6734, because there the file "ru.po" is added
not to the /po directory, but to the root directory.

Дело в том, что когда я отправлял эти pull requests, то ещё плохо
разбирался, как работает Git (да и сейчас не очень уверенно).
Ну так вот. Я сделал глупость. Вместо того чтобы выполнить Fetch и Merge, я
удалил репозиторий у себя и сделал новый Fork.  Соответственно сейчас я не могу
ни обновить эти файлы, ни удалить.
Подскажите, что мне надо сделать. Написать Браму, чтобы он удалил эти файлы? Или
может вы можете их удалить?  И в pull request #6734, так как там вообще файл
"ru.po" добавлен не в каталог /po, а в корневой каталог.

k-takata commented:
> I deleted the repository and made a new Fork.

I don't know how to fix this.
It might be better to close this and #6734 and create new PRs.
I suggest you create 3 PRs:

* Adding NSIS translation.
* Updating ru.po.
* Updating menu files.

NSIS translation is a completely new thing. So, it should be a separate PR.
For ru.po and menu files, please discuss well with @alyoshin. 

Please make sure to use different branches for different PRs.

RestorerZ commented:
@k-takata 
Yes, it's best to close these pull requests (#6762 and #6734).
I will make separate pull requests, as you said, and try not to be so stupid again.


Да, лучше всего закрыть эти pull requests (#6762 и #6734).
Я сделаю отдельные pull requests, как вы и сказали, и постараюсь больше так не глупить.

RestorerZ commented:
due to errors made by the author when creating this pull request

из-за ошибок сделанных автором при создании этого запроса на принятие
